servers/src/main/java/tachyon/worker/netty/NettyDataServer.java

```diff
@@ -71,7 +71,7 @@ public NettyDataServer(final InetSocketAddress address, final BlockDataManager d
   public void close() throws IOException {
     int quietPeriodSecs = mTachyonConf.getInt(Constants.WORKER_NETWORK_NETTY_SHUTDOWN_QUIET_PERIOD);
     int timeoutSecs = mTachyonConf.getInt(Constants.WORKER_NETWORK_NETTY_SHUTDOWN_TIMEOUT);
-    mChannelFuture.channel().close().awaitUninterruptibly();
+    mChannelFuture.channel().close().awaitUninterruptibly(timeoutSecs, TimeUnit.SECONDS);
```

Comment:
left a TODO to investigate this issue.

Integration tests can complete successfully on my machine after this change.